integration_test/third_party_apps_test.go

```diff
@@ -327,6 +328,98 @@ func runSingleTest(ctx context.Context, logger *logging.DirectoryLogger, vm *gce
 	return nonRetryable, nil
 }
 
+// Returns the authoritative set of all apps available for testing.
+// The authoritative list corresponds to the directory names under
+// integration_test/third_party_apps_data/applications
+func determineAllApps(t *testing.T) map[string]bool {
+	allApps := make(map[string]bool)
+
+	files, err := os.ReadDir("third_party_apps_data/applications")
+	if err != nil {
+		t.Fatalf("got error listing files under third_party_apps_data/applications: %v", err)
+	}
+	for _, file := range files {
+		if file.IsDir() {
+			allApps[file.Name()] = true
+		}
+	}
+	return allApps
+}
+
+func modifiedFiles(t *testing.T) []string {
+	cmd := exec.Command("git", "diff", "--name-only", "origin/master")
+	out, err := cmd.Output()
```

Comment:
how do you feel about logging cmd.Output()? I think it would be helpful and not too noisy:
log.Printf("git diff output:\nstdout:%v\nstderr:%v\n", out, err)

particularly for the case where git diff returns an error, this innfo would be great. but I think it would be useful in all cases, if only as a clue that it is runnning git diff and doing stuff with the output in the first place.